src/hotspot/share/opto/intrinsicnode.cpp

```diff
@@ -261,27 +261,37 @@ static const Type* bitshuffle_value(const TypeInteger* src_type, const TypeInteg
   }
 
   if (!mask_type->is_con()) {
-    int mask_max_bw;
-    int max_bw = bt == T_INT ? 32 : 64;
-    // Case 1) Mask value range includes -1.
-    if ((mask_type->lo_as_long() < 0L && mask_type->hi_as_long() >= -1L)) {
-      mask_max_bw = max_bw;
-    // Case 2) Mask value range is less than -1.
-    } else if (mask_type->hi_as_long() < -1L) {
-      mask_max_bw = max_bw - 1;
-    } else {
-    // Case 3) Mask value range only includes +ve values.
-      assert(mask_type->lo_as_long() >= 0, "");
-      jlong clz = count_leading_zeros(mask_type->hi_as_long());
-      clz = bt == T_INT ? clz - 32 : clz;
-      mask_max_bw = max_bw - clz;
-    }
     if ( opc == Op_CompressBits) {
+      int mask_max_bw;
```

Comment:
```suggestion
      // Pattern: Integer/Long.compress(src_type, mask_type)
      int mask_max_bw;
```

Reply:
Can you also say what the meaning of `mask_max_bw` is? Possibly a more expressive name would help here too.

Reply:
If I'm right here, you want to say how many many of the low significance bits are unknown, and that the rest, the high significance bits are all zero, right?

Reply:
You could say that we are trying to find case where we know that the compression result will have at least a certain number of leading zeros, which allows us to restrict the type.

Reply:
It may also be more understandable if you used `min_leading_zeros` instead of `mask_max_bw`.